doc/adrs/070_add_the_possibility_to_hide_or_fade_a_graphic_element.adoc

```diff
@@ -0,0 +1,67 @@
+
+== Decision
+
+We will add to diagram elements two new states: faded and hidden. The first one keep the element visible but more attenuated (transparency or shades of gray). The second one totally hide the graphic element and the layout ignore it. In this case, the representation will propose a way to open a dialog displaying hidden elements with the possibility to make them visible again.
+
+=== Backend
+
+==== Auto-layout
+
+In Sirius Web, the auto-layout is managed by ELK. Sirius gives to ELK the graph and get from ELK a layouted graph. This is this layouted graph that is saved in database. When an element is hidden, we need to keep it in memory to allow the user to display it again. 
+
+The first solution is to totally remove a hidden element from the saved graph and to store it separately. But the element is just hidden, not removed and if a part of the application iterates over all elements of the graph, this part can not access to hidden elements. 
+
+The second solution is to persist these hidden elements after the auto-layout was applied. A way to do that is to indicate to ELK to not consider the element in the layout computation. For this, ELK proposes the https://www.eclipse.org/elk/reference/options/org-eclipse-elk-noLayout.html[`noLayout`] property applies to nodes, border nodes, edges and labels. With this solution hidden elements are still present in the diagram, to invisibilize them we can use CSS.
+
+Concerning faded graphic elements, they is still visible (from the user point of view) so the auto-layout will be applied.
+
+==== Model
+
+The diagram model needs to be evolved to represent these two new states. 
+
+The simplest solution is to add to `Edge` and `Node` an attribute of type `GraphicElementVisibility` that is an enumeration defining `NORMAL`, `FADED` and `HIDDEN`. In this last case, some graphic information like the `position` and `size` doesn't have meaning. Indeed, if the auto-layout is enabled, ELK decides the value of these parameters but it ignores hidden elements (because of the `noLayout`). This can create runtime bugs in the existing code: if a function iterates over graphic elements and uses the position/size for computation, the result can be wrong due to hidden elements. To resolve that, we can make these graphic information attributes optional. The `Optional` is empty if the state is `HIDDEN` and wraps the information otherwise. All code sections requesting these values will throw a compilation error. This makes it easier to identify the code to be adapted and thus avoids hidden execution bugs as described. An other solution is to wrap all graphic information in a common class `GraphicNodeInformation` and `GraphicEdgeInformation` and change existing attributes to a single attribute of these type that is optional. This avoids testing the visibility (or to call `Optional#get`) multiple time if we need to access to several graphic information at the same time.
+
+==== Impacted code
```

Comment:
This is fine but you don't have to try to create an explicit list of all the pieces of code impacted. You could have said, the ELK layout, the incremental layout and this and that.